src/test/java/dev/snowdrop/release/services/IssueServiceTest.java

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ *
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package dev.snowdrop.release.services;
+
+import com.atlassian.jira.rest.client.api.JiraRestClient;
+import com.atlassian.jira.rest.client.api.domain.BasicIssue;
+import com.atlassian.jira.rest.client.internal.async.AsynchronousJiraRestClientFactory;
+import com.fasterxml.jackson.dataformat.yaml.YAMLMapper;
+import dev.snowdrop.release.model.Issue;
+import dev.snowdrop.release.model.POM;
+import dev.snowdrop.release.model.Release;
+import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.test.junit.TestProfile;
+import org.junit.jupiter.api.Test;
+
+import javax.inject.Inject;
+import java.io.InputStream;
+import java.net.URI;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * @author <a href="antcosta@redhat.com">Antonio Costa</a>
+ */
+@QuarkusTest
+@TestProfile(TestProfiles.IntegrationTests.class)
+public class IssueServiceTest {
+
+    private static final YAMLMapper MAPPER = new YAMLMapper();
+
+    @Inject
+    JiraIssueFactory factory;
+
+
+    @Inject
+    IssueService service;
+
+    private static JiraRestClient client(final String jiraUser, final String jiraPw) {
+        AsynchronousJiraRestClientFactory factory = new AsynchronousJiraRestClientFactory();
+        return factory.createWithBasicHttpAuthentication(URI.create(Utility.JIRA_SERVER), jiraUser, jiraPw);
+    }
+
+    @Test
+    public void getIssue() throws Throwable {
+        BasicIssue testIssue = service.getIssue(Issue.TEST_ISSUE_KEY);
+        assertEquals(Issue.TEST_ISSUE_KEY, testIssue.getKey());
+    }
+
+    @Test
+    public void createAndDeleteIssue() throws Throwable {
+        InputStream releaseIS = HelperFunctions.getResourceAsStream("release_template_int_tests.yml");
+        Release releaseObj = MAPPER.readValue(releaseIS, Release.class);
+        releaseObj.setPom(POM.createFrom(HelperFunctions.getResourceAsStream("pom.xml")));
+        BasicIssue issueKey = service.clone(releaseObj,Issue.TEST_ISSUE_KEY,List.of());
```

Comment:
As this test case uses the service which is cloning a JIRA issue like also adding sub-tasks JIRA issues, then the method name should not be `createAndDeleteIssue` but `cloneIssuesAndDeleteThem` !!

I would suggest that instead of testing such a complex use case that we have a simple method doing just `create and delete a jira issue`

Example: 

```
    @Test
    public void createIssue() throws Exception {
        final URI jiraServerUri = new URI("https://issues.redhat.com");

        JiraRestClientFactory restClientFactory = new AsynchronousJiraRestClientFactory();
        JiraRestClient restClient = restClientFactory.createWithBasicHttpAuthentication(jiraServerUri, "yyyyy", "xxxxx");
        IssueRestClient issueClient = restClient.getIssueClient();

        IssueInputBuilder ib = new IssueInputBuilder();
        ib.setIssueTypeId(new Long(3));
        ib.setProjectKey("SB");
        ib.setDescription("This is a dummy test");
        BasicIssue basicIssue = restClient.getIssueClient().createIssue(ib.build()).claim();
        .... TO BE CONTINUED ....
    }
```

Reply:
I have changed the name of the method...but I'm not sure I have to code a new functional method to test something so specific. These tests aim into testing our implemented classes which in this case clone an existing issue and adds some tasks to it. There are still some validations to be made after the issue is created but for this I need the REST API to behave decently. 

WDYT @cmoulliard